### PR TITLE
Single chart - minio TLS

### PR DESCRIPTION
**Pull request by manno**
This refers to #1207

The chart PR is https://github.com/epinio/helm-charts/pull/101

### CODE REVIEW

File: internal/s3manager/s3manager.go [hunk 1]
```diff
@@ -101,7 +111,11 @@ func New(connectionDetails ConnectionDetails) (*Manager, error) {
 // GetConnectionDetails retrieves s3 details from an ini file stored in a
 // secret.
 // Note: The CLI tool in the staging job will use the ini file directly.
+// TODO: Enrico: we removed the NewXXX constructor since it was not used, but this Get was used instead.
```
Comment (manno) [inline internal/s3manager/s3manager.go:114]: TODO

File: internal/s3manager/s3manager.go [hunk 2]
```diff
@@ -126,6 +140,15 @@ func GetConnectionDetails(ctx context.Context, cluster *kubernetes.Cluster, secr
 	}
 	details.Bucket = string(secret.Data["bucket"])
 
+	// load ca cert from minio-tls
+	// if secret is missing just log and skip
+	secret, err = cluster.GetSecret(ctx, "epinio", "minio-tls")
+	if err != nil {
+		logger.Error(err, "error while getting minio-tls secret. Cannot add CA")
+	} else {
+		details.CA = secret.Data["ca.crt"]
```
Comment (manno) [inline internal/s3manager/s3manager.go:149]: we might need 'tls.crt' too?

File: internal/s3manager/s3manager.go [hunk 3]
```diff
@@ -126,6 +135,15 @@ func GetConnectionDetails(ctx context.Context, cluster *kubernetes.Cluster, secr
 	}
 	details.Bucket = string(secret.Data["bucket"])
 
+	// load ca cert from minio-tls
+	// if secret is missing just log and skip
+	secret, err = cluster.GetSecret(ctx, "epinio", "minio-tls")
```
Comment (jimmykarily) [inline internal/s3manager/s3manager.go:140]: for now it's fine but this should be an argument we pass to the epinio server. Maybe the user deployed some S3 compatible solution and they have another secret we should trust instead.

Comment (jimmykarily) [reply internal/s3manager/s3manager.go:140]: Also epinio server shouldn't have hardcoded minio references. It shouldn't care what S3 it's talking to.

File: internal/s3manager/s3manager.go (hunk 2 above)
Comment (jimmykarily) [reply internal/s3manager/s3manager.go]: Added that too.

File: internal/s3manager/s3manager.go (hunk 3 above)
Comment (jimmykarily) [reply internal/s3manager/s3manager.go]: added a setting in values.yaml in the helm chart and an argument in `epinio server` command here: https://github.com/epinio/epinio/pull/1228/commits/ab0c4b71dffcf53a1fe0fcb8fe98bcf21dc88cde

**Comment by jimmykarily** [comment]
Only one test failing for irrelevant reason. I created an issue to fix: https://github.com/epinio/epinio/issues/1235 . We can merge.

**Review by jimmykarily — APPROVED** [review]

**Pull request merged**